### ISSUE TITLE
Bug: await

### ISSUE THREAD

**Issue by makkalay** [typescript-eslint/typescript-eslint#12106]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://github.com/eslint/eslint/issues/20593#issuecomment-4011148917

### Repro Code

```TypeScript
https://github.com/eslint/eslint/issues/20593#issuecomment-4011148917
```

### ESLint Config

```javascript
nuxt 4
```

### tsconfig

```jsonc
nuxt 4 generated
```

### Expected Result

to show that await hare is not needed

### Actual Result

no warning

<img width="782" height="188" alt="Image" src="https://github.com/user-attachments/assets/d2932ea9-ad9a-4ea0-a2d3-74295d66c213" />

**Comment by bradzacher** [comment]
you have provided literally zero context on anything at all.
you've provided a screenshot of code with a syntax error in it and have provided no repro.
you haven't done really any form of searching of our docs to see if there's an existing solution.
you've just linked back to the original eslint issue which had exactly the same lack of content.
in that issue you also tagged random people.

I'm just going to close and lock this as all around you've violated so, so, so many guidelines.
you clearly haven't read anyone's contributor guides to understand best practices and acceptable behaviours.
